Roles, Muting, Nicknaming, Word Filter, Developer Contacting

Skillz4Killz opened the pull request:
1. addRole
2. removeRole
3. mute
4. unmute
5. nickname

Skillz4Killz committed:
added addRole, removeRole, mute, unmute, nickname command pieces

UnseenFaith reviewed (commented):
These look good. However I would recommend you switch to using the Embed constructor instead of inline objects. The Embed constructor will make the flow of the code look more concise and less cluttered.

Skillz4Killz committed:
converted embed to Embed Constructors

Skillz4Killz commented:
Hold on let me try to fix the errors the CI found and ill resubmit it

Skillz4Killz committed:
fixed some of the errors reported by the CI

Skillz4Killz committed:
missed 17 errors but i think i got them now

Skillz4Killz committed:
a couple left but fixed

Skillz4Killz committed:
This should be the final error

Skillz4Killz commented:
Alrighty seems like this is good to go now. All errors should be fixed and ready for you. Hopefully ive gotten everything. Cheers! @UnseenFaith

Skillz4Killz committed:
added naughty word filter monitor

Skillz4Killz committed:
fixed a couple of errors

Skillz4Killz committed:
should have fixed all errors

Skillz4Killz commented:
Added a monitor for filtering words

Skillz4Killz committed:
fixed a infinite loop error

CyberiumShadow commented:
The mute and unmute commands can be combined,

same with the addRole and removeRole commands.

Skillz4Killz commented:
I think they are better off staying separate. Having 1 command makes for a much worse user experience by having users be required to input an additional command. `.mute enable/disable @user` means they have to add an additional input as opposed to just `.mute @user` and `.unmute @user`.

kyranet commented:
No, check if the user is muted, if so, unmute, otherwise remove. Same for add/removeRole

Skillz4Killz commented:
Ooo that is interesting I hadn't thought of that. I will need a day or two but I will definitely make that happen and resend.

Skillz4Killz committed:
converted mute and role commands into 1 each

Skillz4Killz committed:
slight error fixes

Skillz4Killz committed:
couple leftovers errors

Skillz4Killz commented:
1. Converted Mute and Unmute into 1 mute file
2. Converted addRole and removeRole into 1 role file

Let me know if there is anything I missed or else to improve on. :)

kyranet commented:
That's much better, however, you should use Komada's configuration system instead of storing manual/hardcoded variables, so it's more dynamic. As well as avoiding have two blocks almost equal, in the mute command, you should use a ternary, that will save you about +10 lines.

CyberiumShadow commented:
@Skillz4Killz please use Komada's Configuration System for the manually set variables

Skillz4Killz commented:
@kyranet @CyberiumShadow Hey sorry was busy on a project last few days. So 1 main question is how do I use the Config System if there is no config.json file in this repo? Do you guys want me to make a guilds_Schema.json? I don't think that really serves as a good file to have here, but I'll add it if you guys wish.

Or would you guys prefer having the declarations being commented out and just explaining to add these vars in the schema file?

UnseenFaith commented:
@Skillz4Killz 
```js
exports.init = (client) => {
  // do add schema code here
}
```

Skillz4Killz committed:
converted vars to config system and used ternary operator on mute and role commands

Skillz4Killz committed:
fixed some errors

Skillz4Killz committed:
fixed some errors

UnseenFaith commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,58 @@
+exports.run = async (client, msg, [user]) => {
+  try {
+    const value = msg.mentions.members.first().roles.get(muteRoleId);
+    if (msg.mentions.members.first().roles.get(muteRoleId)) {
```
You would need to get the settings again for use in the command. You can use this trick to get all the variables you need.
```js
const { muteRoleId, reportChannelId, titleURL, muteRoleName } = msg.guild.settings
```

UnseenFaith commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,58 @@
+  try {
+    const value = msg.mentions.members.first().roles.get(muteRoleId);
+    if (msg.mentions.members.first().roles.get(muteRoleId)) {
+      msg.mentions.members.first().removeRole(msg.guild.roles.find("name", muteRoleName));
```
This will be fixed the by the above

UnseenFaith commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,58 @@
+      .setFooter("Made with Komada, Discord.js, & <3", client.user.avatarURL)
+      // Takes a Date object, defaults to current date.
+      .setTimestamp()
+      .setURL(titleURL);
```
Fixed by above

UnseenFaith commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,58 @@
+    msg.reply({ embed });
+  } catch (e) {
+    msg.reply("Some error occured with mute/un-muting the member. A report has been sent to the developers.");
+    client.channels.get(reportChannelId).send(`There was an error trying to mute/un-mute: ${e} in ${msg.channel} on ${msg.guild} by ${msg.author}`);
```
Fixed by above

Skillz4Killz committed:
fixed with the const declaration

Skillz4Killz committed:
fixed couple errors

Skillz4Killz commented:
@UnseenFaith Hey sorry to bother u but i changed it as you suggested and its still throwing up 4 errors. These errors are really stupid though. 1 tells me an declared var has to be used. Another tells me that same var was never declared.

ghost commented:
@Skillz4Killz they arent stupid

UnseenFaith commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,58 @@
+};
+exports.run = async (client, msg, [user]) => {
+  try {
+    const { muteRoleId, reportChannelId, titleURL, muteRoleName } = msg.guild.settings;
```
You declared it in the try statement, instead of above it, so it's not available in the catch statement.

ghost commented:
What @UnseenFaith said

UnseenFaith commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,66 @@
+};
+exports.run = async (client, msg, [user, role]) => {
+  try {
+    const { titleURL, reportChannelId } = msg.guild.settings;
```
Same thing here. Declared it in the try instead of outside.

UnseenFaith reviewed (changes requested):
Fix the changes I requested and it should be fine.

ghost commented:
``exports.init = async (client) => {
  const schema = client.settings.guilds.schema;
  if (!schema.muteRoleName) {
    await client.settings.guilds.add("muteRoleName", "MUTEROLENAME");
  }
  if (!schema.muteRoleId) {
    await client.settings.guilds.add("muteRoleId", "MUTEROLEID");
  }
  if (!schema.titleURL) {
    await client.settings.guilds.add("titleURL", "TITLEURL");
  }
  if (!schema.reportChannelId) {
    await client.settings.guilds.add("reportChannelId", "CHANNELIDHERE");
  }
};`` Here its trying to use reportChannelId tho it cant bc u defined reportChannelId in a blocked scope dont quote me on that tho

UnseenFaith commented:
That's wrong. He's definining things to be added to schema at the load of the command, which will be available on the guild settings object.

ghost commented:
Owo i see

UnseenFaith commented:
@Skillz4Killz I suggest you add the schema exports.init code to all commands where you need a variable added to the schema though. That way it ensures the schema key is added even if it doesn't exist, or if they don't use all your commands.

Skillz4Killz committed:
fixed couple errors

Skillz4Killz commented:
You guys are right turns out the errors did make sense and I was the stupid one :)

I believe everything is fixed and ready to go. Let me know if there is anything else that's needed or could be improved.

Skillz4Killz commented:
Anything I can change or add on this?

ghost commented:
Have u done what @UnseenFaith (Faith) asked?

Skillz4Killz commented:
@MrJacz, As far as i know, i did everything Faith mentioned. I'm not sure if I missed anything I've looked through the review and it appears I did do everything.

Skillz4Killz committed:
added a contact feature

Skillz4Killz committed:
fixed a couple errors

Skillz4Killz commented:
Added another piece that gives the ability for a person to contact the bot owners in specific channels in regards to the message type.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,71 @@
+
+exports.run = async (client, msg, [type, ...content]) => {
+  // Declares all necessary variables to use for the command.
+  const { bugChannelId, ideaChannelId, messageChannelId, reportChannelId } = msg.guild.settings;
```
If it's to contact with bot developers, why is per-guild?

Skillz4Killz replied:
I could not find another method to doing so. I thought you guys did not like adding variables on the commands itself. In my bot, these variables are in the config.json file so my bot can use them easily but if there is a better way I would be happy to change them and recommit.

kyranet replied:
If this is going to be a piece, please do not use a config.js/json but a plain channel id.

Also, I would suggest this:
```javascript
exports.channel = null;

exports.init = (client) => {
  this.channel = client.channels.get("CHANNEL_ID");
}
```

And then in the code, you just do `this.channel.send`. It's perfectly fine as Channel will be a [pointer](https://en.wikipedia.org/wiki/Pointer_(computer_programming)) and won't affect memory usage negatively, in fact, it'll perform much better than re-getting this once and again.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,71 @@
+exports.init = async (client) => {
```
To keep consistency with the rest of pieces, please place exports.init below exports.conf and exports.help (just style)

Skillz4Killz replied:
Will move them all to the bottom in all the files in the next commit.

Skillz4Killz committed:
fixed all changes Kyra requested

Skillz4Killz committed:
fixed 2 CI errors

Skillz4Killz commented:
Just finished everything you requested Kyra. Really sorry about forgetting this. I got super busy with stuff and trying to fix my bot and totally forgot to fix this. Sincerest apologies

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+exports.init = async (client) => {
+  const schema = client.settings.guilds.schema;
+  if (!schema.muteRoleName) {
+    await client.settings.guilds.add("muteRoleName", "MUTEROLENAME");
```
muteRoleName is useless, use `muteRole` (Id is redundant) instead.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+    await client.settings.guilds.add("muteRoleName", "MUTEROLENAME");
+  }
+  if (!schema.muteRoleId) {
+    await client.settings.guilds.add("muteRoleId", "MUTEROLEID");
```
I shall link you to the docs, because you are not getting how SchemaManager#add works. https://dirigeants.github.io/komada/tutorial-SettingGateway_.html

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+    await client.settings.guilds.add("muteRoleId", "MUTEROLEID");
+  }
+  if (!schema.titleURL) {
+    await client.settings.guilds.add("titleURL", "TITLEURL");
```
titleURL is also useless, let it be added by the bot's owner.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+    await client.settings.guilds.add("titleURL", "TITLEURL");
+  }
+  if (!schema.reportChannelId) {
+    await client.settings.guilds.add("reportChannelId", "CHANNELIDHERE");
```
Instead of `reportChannelId`, which is a complicated name, use `modlog` instead.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,55 @@
+exports.init = async (client) => {
+  const schema = client.settings.guilds.schema;
+  if (!schema.titleURL) {
+    await client.settings.guilds.add("titleURL", "TITLEURL");
```
Same here and below.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,67 @@
+exports.init = async (client) => {
+  const schema = client.settings.guilds.schema;
+  if (!schema.titleURL) {
+    await client.settings.guilds.add("titleURL", "TITLEURL");
```
Same here and below.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,40 @@
+  try {
+    // Determines which type of contact and sends to the appropriate channel.
+    if (type === "bug") {
+      this.channel.send(`Bug Report From: ${msg.author} in ${msg.guild}\n\n ${content.toString().replace(/,/g, " ")}`);
```
Reduce the code repetition and redundancy with a single `this.channel.send`. Another thing I have noticed, this command runs in any channel, even in DMs, `msg.guild` will output `null` in the latter case.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+exports.run = async (client, msg, [user]) => {
+  const { muteRoleId, reportChannelId, titleURL, muteRoleName } = msg.guild.settings;
+  try {
+    const value = msg.mentions.members.first().roles.get(muteRoleId);
```
Why are you parsing this manually? I shall point you to the docs. https://komada.js.org/command-arguments.html

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+  try {
+    const value = msg.mentions.members.first().roles.get(muteRoleId);
+    if (msg.mentions.members.first().roles.get(muteRoleId)) {
+      msg.mentions.members.first().removeRole(msg.guild.roles.find("name", muteRoleName));
```
Thought you'll remove the `msg.mentions.members.first()` with the previous requested change, that's a lot of duplicated code.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+exports.run = async (client, msg, [user]) => {
+  const { muteRoleId, reportChannelId, titleURL, muteRoleName } = msg.guild.settings;
+  try {
```
Why do you wrap everything in a try/catch? That slows the command a lot, any rejected promise will be caught by the command handler.

Skillz4Killz replied:
try/catch allows me to send the errors to a specific channel instead of having to read the logs. Is there an easier way to do this? When im not on computer i can see the errors appear on my #error-log and easily know what went wrong

kyranet replied:
The errors you'll get with that code are mostly permissions related, nothing you can fix unless you check them beforehand and efficiently.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,59 @@
+exports.conf = {
+  enabled: true,
+  runIn: ["text"],
+  aliases: ["unmute", ""],
```
That empty alias might break some stuff.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,55 @@
+exports.run = async (client, msg, [user, name]) => {
+  const { reportChannelId, titleURL } = msg.guild.settings;
+  try {
```
Same here (try/catch)

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,55 @@
+  runIn: ["text"],
+  aliases: ["nick"],
+  permLevel: 2,
+  botPerms: ["CHANGE_NICKNAME", "MANAGE_NICKNAMES"],
```
When changing the bot's nickname, it doesn't require MANAGE_NICKNAMES, but only CHANGE_NICKNAME. The second one is to change somebody else's nickname.

Skillz4Killz replied:
That's what the command is. It is for changing another users nickname. For example, I have people join the server with really filthy names and the mods can use this command to edit their nickname.

I was just thinking that if the user also wanted to change bots nickname they would need CHANGE perm also

kyranet replied:
Cannot confirm, but I'm pretty sure MANAGE_NICKNAMES overrides CHANGE_NICKNAME

tjrgg replied:
Can confirm.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,67 @@
+exports.run = async (client, msg, [user, role]) => {
+  const { titleURL, reportChannelId } = msg.guild.settings;
+  try {
+    const value = msg.mentions.members.first().roles.find("name", role);
```
Same here (manual argument parsing), and extreme code duplication.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,67 @@
+exports.run = async (client, msg, [user, role]) => {
+  const { titleURL, reportChannelId } = msg.guild.settings;
+  try {
```
This try/catch.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,40 @@
+exports.run = async (client, msg, [type, ...content]) => {
+  try {
```
Again, try/catch.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,32 @@
+  // Do not delete messages from the Bot itself otherwise it will run on a loop when it sends msgs in a DM to author or reports to a channel.
+  if (msg.author.bot) return;
+  // Add your list of words you want to filter. This is a very powerful filter so be careful of using. For example, the word 'assign' will not be allowed if the word `ass` is present in the swear words. In other filters, they are easily breakable by typing 'ass.' but this filter will delete these types of breaks.
+  const swearWords = ["damn", "shucks", "frak", "ass"];
```
This must be configureable in the guild settings. Refer [SettingGateway examples](https://dirigeants.github.io/komada/tutorial-SettingGateway_.html#) to know how.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,32 @@
+
+exports.run = (client, msg) => {
+  // Do not delete messages from the Bot itself otherwise it will run on a loop when it sends msgs in a DM to author or reports to a channel.
+  if (msg.author.bot) return;
```
Don't put this, just change `ignoreBots` to true.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,32 @@
+      badWordList += badWords[i];
+    }
+    // DM the user regarding what words caused the message to be deleted.
+    msg.author.send(`I deleted the message below because you used a word that is not allowed. : **${badWordList}**`);
```
badWordList is an array, stringified, it gives a string with all values joined by comma. For formatting, you might want to add `.join(", ")`.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,32 @@
+    // DM the user regarding what words caused the message to be deleted.
+    msg.author.send(`I deleted the message below because you used a word that is not allowed. : **${badWordList}**`);
+    // DM the user with the message incase they wish to copy paste -> edit and resend. Imagine what happens when by accident a user gets the message deleted for a mistake.
+    msg.author.send(`${msg.content}`);
```
Instead of creating a string out of another string: `msg.author.send(msg.content);`.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,32 @@
+    // DM the user with the message incase they wish to copy paste -> edit and resend. Imagine what happens when by accident a user gets the message deleted for a mistake.
+    msg.author.send(`${msg.content}`);
+    // Delete a message
+    msg.delete()
```
You should check if the bot has permissions to delete messages (msg.deletable) before processing everything.

kyranet reviewed (changes requested):
A lot of requested changes in regards to duplicated/bad quality/redundant code, found some references to GuideBot which are deprecated in Komada (as we have internal argument parsing).

Also, don't put everything into embeds, keep embeds for modlogs, not responses, a lot of servers disable the EMBED_LINKS permissions, and not every user has embed links enabled in their ends.

Skillz4Killz committed:
Fixed all requests made by Kyra

Skillz4Killz committed:
forgot to save nickname file

Skillz4Killz committed:
fixed a couple of indent and semi colon errors

Skillz4Killz commented:
@kyranet I have fixed everything you stated. let me know if there is anything else i can do to improve it

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+exports.run = async (client, msg, [user]) => {
+  const { muteRole } = msg.guild.settings;
+  const value = user.roles.find("name", muteRole);
```
muteRole is either `null` or a role ID. Use get instead.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  const { muteRole } = msg.guild.settings;
+  const value = user.roles.find("name", muteRole);
+  if (value) {
+    user.removeRole(msg.guild.roles.find("name", muteRole));
```
Same here and 2 lines below. And use `await` to avoid raceconditions and uncaught promises.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  } else {
+    user.addRole(msg.guild.roles.find("name", muteRole));
+  }
+  msg.reply(value ? `${user} is is no longer in the time-out corner. :smiley:` : `${user} is now in the time-out corner. :smiley:`);
```
Instead of using the emoji's representation in text, use unicode instead. You can do that by writting `\:smiley:`. Also, the message is a bit informal (it made me check the commands name).

Besides that, put a `return`.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  runIn: ["text"],
+  aliases: ["unmute"],
+  permLevel: 2,
+  botPerms: ["MUTE_MEMBERS"],
```
MUTE_MEMBERS is a voice channel permission. You want MANAGE_ROLES.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,29 @@
+exports.run = async (client, msg, [user, name]) => {
```
You should put name as optional, defaulting to `""`, so the bot can clear nicknames.

tjrgg replied:
```[user, name = ""]```

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,29 @@
+exports.run = async (client, msg, [user, name]) => {
+  // Checks to see if the nickname is too long or too short for the discord requirement if so replies and then exits.
+  if (name.length > 32 || name.length === 0) {
```
And remove these checks as the parameter requires a string of 1-32 chars.

CyberiumShadow replied:
Pre sure that is checking if name.length is **greater** than 32 or is 0 to make a warning that the nickname is too long or short.

kyranet replied:
No, that'll never trigger the check if the message is 0 chars long or more than 32, check usage.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,29 @@
+  // Set the nickname
+  user.setNickname(name);
+  // Send a reply saying it is done
+  msg.reply(`Nickname has been changed to ${name} for ${user} :smiley:`);
```
Same here (unicode).

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,29 @@
+    return;
+  }
+  // Set the nickname
+  user.setNickname(name);
```
Same here (await).

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,38 @@
+    // Check if the user already has the role
+    if (user.roles.find("name", role)) {
+      // Remove role if already posses
+      user.removeRole(msg.guild.roles.find("name", role));
```
Same here (await).

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,38 @@
+      // Remove role if already posses
+      user.removeRole(msg.guild.roles.find("name", role));
+      // Send reply
+      msg.reply(`${user} has lost the ${role} role. :smiley:`);
```
Same here (return message).

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,38 @@
+  enabled: true,
+  runIn: ["text"],
+  aliases: ["addrole", "ar", "removerole", "rr"],
+  permLevel: 2,
```
Due to the nature of this command, you should change the level to 3. What if the bot's role is above everyone and a moderator asks the bot to grant them the administrator role? This command has (certainly) a lot of flaws, a set of configured roles (claimable roles) would be more useful in most situations.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,37 @@
+exports.run = async (client, msg, [type, ...content]) => {
+  // Determines which type of contact and sends to the appropriate channel.
+  let output = "";
+  if (type === "bug") {
```
What about a switch? More performant and cleaner. And don't use `+=` but `=` (you're not appending that string into the string.

Additionally, a shorter and performant solution would be:

```javascript
const types = {
  bug: "Bug Report",
  idea: "New Idea",
  message: "New Message",
};

// ...

return this.channel.send(`${types[type]} ...`);
```

CyberiumShadow replied:
Talking with Gus about this, having 3 pathways for the if else and having it in a switch case doesn't make much of a noticeable difference in performance, but if Kyra suggests the switch case, then go ahead.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,37 @@
+  } else {
+    output += "New Message";
+  }
+  this.channel.send(`${output} From: ${msg.author} in ${msg.guild}\n\n ${content.toString().replace(/,/g, " ")}`);
```
Same here (return message).

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,37 @@
+  permLevel: 0,
+  botPerms: [],
+  requiredFuncs: [],
+  cooldown: 0,
```
You might want to put a cooldown, as people would be able to spam your channel.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+  // Change to false if the filter should not ignore other bots
+  ignoreBots: true,
+  // Change to true if the filter should ignore selfBots
+  ignoreSelf: false,
```
Well, unless you have an echo command, you might not want your own bot to check its own messages.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+
+exports.run = (client, msg) => {
+  // Please add the channel ID you want to recieve these reports in.
+  const modlog = "CHANNELIDHERE";
```
Use the modlog configurable settings.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+    if (badWords.length > 0) {
+      let badWordList = [];
+      for (let i = 0; i < badWords.length; i++) {
+        badWordList += badWords[i];
```
No, no, that's not how you push values into an array, you use [Array.prototype.push](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/push)

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+  }
+};
+
+// Create a swear words setting for every guild 
```
Trailing space over here.

CyberiumShadow replied:
eslint reee

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+        badWordList += badWords[i];
+      }
+      // DM the user regarding what words caused the message to be deleted.
+      msg.author.send(`I deleted the message below because you used a word that is not allowed. : **${badWordList.join(", ")}**`);
```
Again, await. Also, you might want to send that in modlogs, not notify the user.

Skillz4Killz replied:
This has to be to the user because it allows the user to know what message was deleted and the cause of the deletion. For example, I send a 1500 word announcement message but it has a bad word in there i didn't intend, it's good to receive a message saying hey this was deleted for this reason and here is the message so i can copy paste the deleted message and edit the necessary words and send it again

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+  // Please add the channel ID you want to recieve these reports in.
+  const modlog = "CHANNELIDHERE";
+  if (msg.deletable) {
+    const badWords = msg.guild.settings.swearWords.filter(word => msg.content.toLowerCase().includes(word));
```
You might want to check if there's **at least** one element in the array.

kyranet reviewed (changes requested):
You improved, however, another 20 requested changes.

tjrgg commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,29 @@
+  description: "Change nickname of a user",
+  usage: "<user:member> <name:str{1,32}>",
+  usageDelim: " ",
+  extendedHelp: "1) User will require a role called Moderators\n2) Bot will require a role higher than the user to be able to change nickname.\n3) Up to a maximum of 3 words with spaces and max characters must be 32 or less.",
```
Same here (modRole instead of "Moderators").

tjrgg commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,38 @@
+  if (msg.guild.roles.exists("name", role)) {
+    // Check if the user already has the role
+    if (user.roles.find("name", role)) {
+      // Remove role if already posses
```
Double check your spelling.

tjrgg commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,38 @@
+  description: "Assign a role to another user.",
+  usage: "<user:member> <role:str>",
+  usageDelim: " ",
+  extendedHelp: "1) User must have a role called Moderators to use this command.\n2) Bot must have Manage Role permissions. The bot will not be able to assign a role higher than its highest role.",
```
Same here (modRole instead of "Moderators").

tjrgg commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,37 @@
+
+exports.init = (client) => {
+  // Insert channel ID of the channel you would like for the users to contact you on.
+  this.channel = client.channels.get("CHANNEL_ID");
```
Maybe use a config option for this? I feel that would be better than hardcoding a channel ID in and would make it easy to change later, if necessary.

kyranet replied:
I agree: `client.config.supportChannel` 👍

Skillz4Killz replied:
Would you guys mind sending me a link to the doc page where I can learn to add a config option? I've been looking through it and the only options I can add is in settings gateway but that's per guild.

UnseenFaith replied:
Any variables passed to Komada's startup will be available on `client.config`
Example:
```js
new Komada({
  prefix: blah,
  ..more here,
  myDefaultSupportChannel: 'someID',
  }).login(token);

// You now have the channel ID available from `client.config.myDefaultSupportChannel`

Skillz4Killz replied:
But how do I pass a variable to already existing Komada startup in a Komada piece? this piece has to be standalone and i have no way to force a user to edit their komada configurations. @UnseenFaith

tjrgg commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,42 @@
+  if (!schema.swearWords) {
+    await client.settings.guilds.add("swearWords", { type: "String", array: true });
+  }
+};
```
Add help information to this monitor to be consistent with the other monitors.

tjrgg commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  description: "Mutes/unmutes a person on both text and voice.",
+  usage: "<user:member>",
+  usageDelim: "",
+  extendedHelp: "1) mute @user\n2) Requires the user to have a role called Moderators\n3) Bot requires Mute Members permissions.\n4) Requires a role that is called 'Time-Out' set up without any permissions and at a high level in the role settings as well as each channel permissions being edited with its settings.",
```
Modify this to say that the user will require the modRole. It may not necessarily be called "Moderators".

tjrgg reviewed (changes requested):
Requested some minor changes after finding some inconsistencies and things I thought could be improved. In addition, I'd recommend looking through all your comments just to ensure everything makes sense. I had to read through some of your comments a couple times to understand what was being said. I'd maybe even consider removing some of them altogether. Looking through the other pieces in this repo, I noticed most of them don't have a lot of comments and that's simply because they aren't necessary. Comments should only be included when special notes need to be made about compatiability or if it would be difficult for the developer to understand what is going on, imo. I don't believe most of your comments fall under either one of those categories. That being said, they don't do too much harm, so if one of the collaborators doesn't have a problem with them, that's cool too.

Skillz4Killz committed:
fixed more requested changes

Skillz4Killz committed:
removed unnecessary else {}

Skillz4Killz committed:
missing semicolon

Skillz4Killz committed:
unnecessary else {} after return

Skillz4Killz committed:
added async to make await work

Skillz4Killz committed:
removed extra schema declaration

Skillz4Killz committed:
trying to resubmit because travis broke

Skillz4Killz committed:
changing let to const

Skillz4Killz commented:
Let me know if there are any other things that can be improved as well please. Cheers

kyranet commented:
There are a few things. I'll list them tomorrow 👍

tjrgg commented on monitors/filterWords.js:
```diff
@@ -1,42 +1,44 @@
-      let badWordList = [];
-      for (let i = 0; i < badWords.length; i++) {
-        badWordList += badWords[i];
+    if (msg.guild.settings.swearWords) {
```
I believe this an array, correct? If that's the case, you should check to see if the array as something in it (aka a word), in addition to checking if the setting exists.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  const { muteRole } = msg.guild.settings;
+  const value = user.roles.get(muteRole);
+  if (value) {
+    await user.removeRole(msg.guild.roles.get(muteRole));
```
You should check if the role exists beforehand.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+  } else {
+    await user.addRole(msg.guild.roles.get(muteRole));
+  }
+  return msg.reply(value ? `${user} is is no longer muted. 😄` : `${user} is now muted. 😄`);
```
is is no, you could also do:

```javascript
`${user} is ${value ? "not " : ""} longer muted.`;
```

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,35 @@
+exports.init = async (client) => {
+  const schema = client.settings.guilds.schema;
+  if (!schema.muteRole) {
+    await client.settings.guilds.add("muteRole", { type: "String" });
```
type Role.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,29 @@
+exports.run = async (client, msg, [user, role]) => {
+  if (msg.guild.roles.exists("name", role)) {
```
Collection#exists followed by Collection#find kills the performance. Find it and check its existence. Also, you might want to make it case insensitive.

UnseenFaith replied:
Use Komada's Role usage instead, then all you need to do is check `if (role)`

Skillz4Killz replied:
I think both are needed no?
First, it checks if the role exists on the server. 
Then it checks if the user has the role: A) if yes -> remove the role B) if it doesn't have added role
third, if the role doesn't exist on the server send a message saying role doesn't exist

If i merge it into 1 if statement to check server and role ill have to make multiple if statements
1) Check if role exists && user has role
2) check if role exists && user doesnt have role
3) if role doesnt exist send error message

kyranet replied:
I mean, you could just use Komada's internal parser for the role type, which resolves mention, id, and iirc, also names. 

Map.prototype.has is dozen times faster than `Collection.prototype.exists` and therefore `Collection.prototype.find`.

You get the role object, then check if the user **has** the role, if so, give the role. 

What you're doing is checking a roles collection and get a Boolean from there (`Collection.prototype.exists`), then check if the user has said role with the same name, this is unaccurate as there can be two roles with the same name, and in Guild.roles it'd get one role but in GuildMember.roles get another. 

Then get the same value you have checked, for second time, to apply the role to the user. 

You're doing three checks, twice for the same thing in the same collection. 

The preferable step is to make Komada get the role internally throught the parser, you know 100% it exists, and Komada can use cmdPrompt in case you entered a wrong role name, as opposed to writting it again (or editing it if cmdEditing is set to true), check if the GuildMember has said role by checking the ID, which is extremely fast, and add the role straight-up from there, without fetching it again from the collection.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,29 @@
+      return msg.reply(`${user} has lost the ${role} role. 😄`);
+    }
+    await user.addRole(msg.guild.roles.find("name", role));
+    return msg.reply(`${user} has been given the ${role} role. 😄`);
```
You could remove this duplication like you did in the mute command.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,29 @@
+exports.run = async (client, msg, [user, role]) => {
+  if (msg.guild.roles.exists("name", role)) {
+    if (user.roles.find("name", role)) {
```
Once you do that, you'll be able to use `user.roles.has(role.id);`

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,33 @@
+    idea: "New Idea",
+    message: "New Message",
+  };
+  return client.config.supportChannel.send(`${types[type]} From: ${msg.author} in ${msg.guild}\n\n ${content.toString().replace(/,/g, " ")}`);
```
As you said in the documentation you just write for this command, supportChannel is a **id**. You shall redo the previous step with `exports.init` + `exports.channel`.

Also, your method will replace legit commas into spaces. `...content` is an Array, use [Array.prototype.join](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join) instead.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+exports.run = async (client, msg) => {
+  const modlog = msg.guild.settings.modLog;
+  if (msg.deletable) {
```
Instead of making a christmas tree, remove some level of indentations with negative check + return.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+exports.run = async (client, msg) => {
+  const modlog = msg.guild.settings.modLog;
+  if (msg.deletable) {
+    if (msg.guild.settings.swearWords) {
```
Are you aware that swearWords, when inconfigured, is an empty array, and that `[]` is truthy? Use [Array.prototype.length](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/length) instead.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+      const badWords = msg.guild.settings.swearWords.filter(word => msg.content.toLowerCase().includes(word));
+      if (badWords.length > 0) {
+        const badWordList = [];
+        for (let i = 0; i < badWords.length; i++) {
```
Why this? badWords is a perfectly "workable" array.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+        await msg.author.send(`I deleted the message below because you used a word that is not allowed. : **${badWordList.join(", ")}**`);
+        await msg.author.send(msg.content);
+        await msg.delete()
+          .then(client.channels.get(modlog).send(`Deleted message from ${msg.author} that said the \n\n **Naughty Words:** ${badWordList} in the server ${msg.guild} in the channel ${msg.channel}. \n\n **Full Message:** ${msg.content}`))
```
`client.channels.get` is a no-no for most cases. Use `msg.guild.channels.get` instead. Modlog channels are always in the same guild the command got executed.

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+      }
+    }
+  } else {
+    const serverOrDM = msg.guild ? `Server: ${msg.guild}` : "Message was sent in a DM"
```
This check makes no sense, honestly. If it's in DMs... where's the modlog channel at?

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,44 @@
+exports.run = async (client, msg) => {
+  const modlog = msg.guild.settings.modLog;
```
You should filter DMs.

kyranet reviewed (changes requested):
Better now. 13 more requested changes.

Skillz4Killz committed:
fixing requested changes

Skillz4Killz committed:
fixed some CI errors

Skillz4Killz commented:
I've done all the changes except the change to the usage role.

It's not possible to use that for this commands.

Let me know if there is anything else :)

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+exports.run = async (client, msg, [user]) => {
+  const { muteRole } = msg.guild.settings;
+  if (!msg.guild.settings.muteRole) return msg.reply("Sorry you have not created a mute role for this server.");
```
If you have defined muteRole from msg.guild.settings, why are you getting it again?

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+  const { muteRole } = msg.guild.settings;
+  if (!msg.guild.settings.muteRole) return msg.reply("Sorry you have not created a mute role for this server.");
+  const value = user.roles.get(muteRole);
+  if (value && msg.guild.roles.get(muteRole)) {
```
Would be better if you used Collection#has instead. However, if the member has the role, the guild itself has the role aswell. So this check is redundant.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+exports.run = async (client, msg, [user]) => {
```
`user` is a very confusing variable name for a GuildMember instance. Use member instead.

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+exports.run = async (client, msg, [user]) => {
+  const { muteRole } = msg.guild.settings;
+  if (!msg.guild.settings.muteRole) return msg.reply("Sorry you have not created a mute role for this server.");
+  const value = user.roles.get(muteRole);
```
What if you get the role from Guild.roles, and check if the user has said role by id?

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+  } else {
+    await user.addRole(msg.guild.roles.get(muteRole));
+  }
+  return msg.reply(`${user} is ${value ? "no longer" : "now"} muted. 😄`);
```
not longer*

kyranet replied:
Also, `msg.send`

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,36 @@
+};
+
+exports.init = async (client) => {
+  const schema = client.settings.guilds.schema;
```
If you're checking a single value from the schema, this variable does not make sense anymore.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,22 @@
+exports.run = async (client, msg, [user, name = ""]) => {
+  await user.setNickname(name);
+  return msg.reply(`Nickname has been changed to ${name} for ${user} 😄`);
```
If the nickname is empty, it'll reset. That'll make it output this:

```javascript
Nickname has been changed to  for @Skill 😄
```

You might want to check if `name.length === 0`, if so, say nickname cleared. Otherwise what this command already says.

kyranet replied:
Also, `msg.send`

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,22 @@
+exports.run = async (client, msg, [user, name = ""]) => {
+  await user.setNickname(name);
```
You might want to check if the selected member has higher/equal role position than the author, so the bot doesn't change the nickname to users with a role higher than yours.

kyranet commented on commands/Moderation/nickname.js:
```diff
@@ -0,0 +1,22 @@
+exports.run = async (client, msg, [user, name = ""]) => {
```
Same here (variable 'user')

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,27 @@
+exports.run = async (client, msg, [user, role]) => {
+  if (!msg.guild.roles.exists("name", role)) return msg.reply("There is no role by that name in this server.");
```
You can use Roles#find instead, so you don't retrieve the role by its name twice.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,27 @@
+exports.run = async (client, msg, [user, role]) => {
+  if (!msg.guild.roles.exists("name", role)) return msg.reply("There is no role by that name in this server.");
+  if (user.roles.find("name", role)) {
```
Assign this as a constant, and then in the message sent, use it. Instead of fetching the role for third time.

kyranet commented on commands/Moderation/role.js:
```diff
@@ -0,0 +1,27 @@
+  } else {
+    await user.addRole(msg.guild.roles.find("name", role));
+  }
+  return msg.reply(`${user} has ${user.roles.find("name", role) ? "lost" : "been given"} the ${role} role. 😄`);
```
`msg.send` here.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,32 @@
+exports.run = async (client, msg, [type, ...content]) => {
+  const types = {
```
Put this object **out** of the exports.run, to create the object only once.

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,32 @@
+    idea: "New Idea",
+    message: "New Message",
+  };
+  return this.channel.send(`${types[type]} From: ${msg.author} in ${msg.guild ? msg.guild : "private message"}\n\n ${content.toString().join(" ")}`);
```
What about you reply the user after the message has been sent?

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,32 @@
+exports.channel = null;
+
+exports.init = (client) => {
+  this.channel = client.channels.get("CHANNEL_ID");
```
As we said before, you could replace `"CHANNEL_ID"` to `client.config.supportChannel`

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,36 @@
+exports.run = async (client, msg) => {
+  if (!msg.guild) return;
+  const modlog = msg.guild.settings.modLog;
+  if (msg.deletable && msg.guild.settings.swearWords.length > 0) {
```
Negative check here, you're making a Christmas tree.

kyranet reviewed (changes requested):
18 more requested changes.

CyberiumShadow commented:
1 and a half months of this. 

I will be closing in 2 days if no new changes are added

Skillz4Killz committed:
fixed changes kyra requested EXCEPT last 2 not sure how to do and the grammar change on 'not longer'

Skillz4Killz committed:
small fix to filterWord to fix christmas tree

Skillz4Killz committed:
fixing CI errors

Skillz4Killz committed:
fixed muteRole error i think

Skillz4Killz committed:
grammar mistake

MrJacz committed:
Here

Skillz4Killz committed:
changed files to jacz code. HUGE thanks to him for helping out

Skillz4Killz committed:
Update mute.js

MrJacz committed:
Update mute.js

Skillz4Killz committed:
Merge branch 'master' into patch-1

Skillz4Killz committed:
Merge pull request #2 from MrJacz/patch-2

Update mute.js

MrJacz committed:
Update mute.js

MrJacz committed:
Update role.js

MrJacz committed:
Update filterWords.js

Skillz4Killz committed:
Merge pull request #5 from MrJacz/patch-5

Update filterWords.js

Skillz4Killz committed:
Merge pull request #4 from MrJacz/patch-4

Update role.js

Skillz4Killz committed:
Merge pull request #3 from MrJacz/patch-3

Update mute.js

Skillz4Killz committed:
Merge branch 'master' into patch-1

Skillz4Killz committed:
Merge pull request #1 from MrJacz/patch-1

Here

Skillz4Killz committed:
Update role.js

Skillz4Killz committed:
Update filterWords.js

Skillz4Killz commented:
Kyra really sorry about the delay. Changed everything u requested and Ao and Jacz helped a lot so huge props to them

kyranet commented on commands/Moderation/mute.js:
```diff
@@ -0,0 +1,28 @@
+exports.run = async (client, msg, [member, role]) => {
+  role = typeof role === "string" ? msg.guild.roles.find("name", role) : role;
```
This makes no sense to me, it's a mute command, not a giverole/removerole command. Use SettingGateway instead.

Skillz4Killz replied:
What I was doing: to mute someone you have to create a role and assign/remove that specific role. This role is placed at the top of hierarchy and each channel has this role muted.

is there a better way to do this?

kyranet replied:
The complete solution is extremely complex, you're required to store modlogs capable of storing "extraData" (in this case, array of role ids) so when you mute somebody, you take out all the roles, and when you do unmute, you retrieve the role ids and give all them back.

Just use SettingGateway to store the Muted role's ID.

Skillz4Killz replied:
@kyranet I could have sworn that's what I had before in mute command. But I was told this way is better and to replace it with this method

I had used setting gateway to have a muteRole per guild

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,34 @@
+  if (badWords.length > 0) {
+    await msg.author.send(`I deleted the message below because you used a word that is not allowed. : **${badWords.join(", ")}**`);
+    await msg.author.send(msg.content);
+    msg.delete().then(m => m.guild.channels.get(modLog).send(`Deleted message from ${m.author.tag} that said the \n\n **Naughty Words:** ${badWords} in the server ${m.guild.name} in the channel ${m.channel.name}. \n\n **Full Message:** ${m.content}`))
```
You forgot a return. And put the `then` in the next line. Also, you're not checking the existence of the modLog key (it could be null and error).

kyranet commented on monitors/filterWords.js:
```diff
@@ -0,0 +1,34 @@
+  const { modLog, swearWords } = msg.guild.settings;
+  if (!msg.deletable && !swearWords.length > 0) return;
+  const badWords = swearWords.filter(word => msg.content.toLowerCase().includes(word));
+  if (badWords.length > 0) {
```
Another indentation that could be removed.

Skillz4Killz replied:
Fixed but question, if indentations bad for any reason? performance possibly?

Pandraghon replied:
Better code readability

Skillz4Killz replied:
Thank you will keep it in mind for future

kyranet commented on commands/Tools/contact.js:
```diff
@@ -0,0 +1,34 @@
+  message: "New Message",
+};
+exports.run = async (client, msg, [type, ...content]) => {
+  this.channel.send(`${types[type]} From: ${msg.author} in ${msg.guild ? msg.guild : "private message"}\n\n ${content.toString().join(" ")}`);
```
await this, a failure will make it throw an uncaught error.
Also, `content.toString()` will join the words by `,`, and String.prototype.join() is not a method as far as I am aware of.

kyranet reviewed (changes requested):
Just, 4 more requested changes. (And please, spacing between multiple conditions would be nice aswell, having everything too compact makes it harder to read)

Skillz4Killz commented:
will try to do these tmrw

kyranet commented:
Are we waiting for Github's Hackathon in order to this PR to get finished someday?

Skillz4Killz committed:
fixed contact

Skillz4Killz committed:
fixed filterwords

kyranet commented:
Will we merge this PR someday?

Stitch07 commented:
@kyranet I can fix it

Skillz4Killz commented:
@kyranet @Soumil07 I have this almost completed ever since that day I left the msgs on it. I am just waiting for Kyra to let me know how to fix the mute problem he found because not sure if it's possible another way without roles. I didn't think I should send it up incomplete so I was waiting to hear back

Skillz4Killz commented:
@Soumil07 Ill send up what I did later on can you just help fix the Mute for me please

Stitch07 commented:
@Skillz4Killz Why dont you use ```msg.channel.overwritePermissions(member, {
      SEND_MESSAGES: false,
    });```

Skillz4Killz commented:
@Soumil07 I am a noob and didn't know that was possible :) but that is an amazing idea

kyranet commented:
Don't overwrite the permissions for each member in each channel, that leads to extreme API spam. Instead, use a configured muted role.

tech6hutch commented:
"I am just waiting for Kyra to let me know how to fix the mute problem **she** found"
he* 😛

Skillz4Killz commented:
Sincerest apologies ;(

Stitch07 commented:
@kyranet but if you use muteroles, the mute is global not channel specific.

kyranet commented:
🎉 Exactly, @Soumil07. What's the point of it if it didn't?

Stitch07 commented:
Most mutes are channel mutes. I want him muted because he didn't follow the channel rules or posted in the wrong channel.

tech6hutch commented:
FWIW, I've never seen a mute (from bot or otherwise) be channel-specific. The mods just mute someone for the whole server.

Skillz4Killz committed:
linter

Skillz4Killz commented:
@Soumil07 Would you mind taking a look at the mute and helping fix that, I am so confused which way is wanted to do this. Would really appreciate it

Stitch07 commented:
I'll take a look. :thumbsup:

Stitch07 commented:
Skimmed this convo. May be wrong, but according to Kyra, you should just set a designated role with SG. So here, that'd be ```if (!client.settings.guildd.scheme.muteRole) { await client.settings.guilds.add("muteRole", { type: "Role" }) //add to the schema if it doesn't exist
const role = msg.guild.roles.get(msg.guild.settings.muteRole)
await member.addRole(role) ;```

This is very rough. You'd have to do a few checks like if the role exists and is addable, etc.

MrJacz commented:
@Soumil07 Depending where u put that code pretty much yes

kyranet commented:
Fix ESLint (check Codacy report) and fix that last requested change.

This PR will get closed on Monday if no changes are made.

(Really, this PR takes a while to load in Chrome, it's too long, and it's time to merge or close this)

CyberiumShadow committed:
Merge branch 'master' into master

CyberiumShadow commented:
This PR has reached a 2month+ lifetime with no substantial progress.

Change deadline has past as well.

PR will be closed, feel free to reopen a new PR once everything is Polished.
